src/awkward/highlevel.py

```diff
@@ -1325,6 +1325,14 @@ def __array__(self, *args, **kwargs):
         with ak._errors.OperationErrorContext("numpy.asarray", (self, *args), kwargs):
             return ak._connect.numpy.convert_to_array(self._layout, args, kwargs)
 
+    def __arrow_array__(self, *args, **kwargs):
+        with ak._errors.OperationErrorContext(
+            f"{type(self).__name__}.__arrow_array__", (self, *args), kwargs
```

Comment:
Without this, the error message would show up in `ak.to_arrow`, since you're calling the high-level version of that and not its `_impl`. But users may be confused because they don't have any `ak.to_arrow` in their own code, so this helps.